app/both/collections/residencies.js

```diff
@@ -137,3 +145,10 @@ Residencies.after.remove(function (userId, residency) {
     entityId: residency._id,
   })
 });
+
+function validateDate(moveIn, moveOut){
```

Comment:
Add a docstring describing the validation rules in the function.

Reply:
Basically, I have to think pretty hard to understand this function, so it may not be self-documenting. Please strive for clarity of code, by using meaningful variable names and adding inline documentation.

Reply:
Sure I will take care of it now onwards

Reply:
Thanks. I don't want to seem too critical, just trying to make it easier to maintain the code. I have certainly written some code that is difficult to understand as well, so feel free to ask for clarifications on any difficult code you find :-)